gulpfile.js

```diff
@@ -0,0 +1,203 @@
+    ],
+    'demostudio_devderby_landing': [
+        'media/js/demos-devderby-landing.js'
+    ],
+    'jquery-ui': [
+        'media/js/libs/jquery-ui-1.10.3.custom/js/jquery-ui-1.10.3.custom.min.js',
+        'media/js/moz-jquery-plugins.js'
+    ],
+    'tagit': [
+        'media/js/libs/tag-it.js'
+    ],
+    'search': [
+        'media/js/search.js',
+        'media/js/search-navigator.js'
+    ],
+    'framebuster': [
+        'media/js/framebuster.js'
+    ],
+    'syntax-prism': [
+        'lib/js/prism/prism.js',
+        'media/js/prism-mdn/components/prism-json.js',
+        'media/js/prism-mdn/plugins/line-numbering/prism-line-numbering.js',
+        'lib/js/prism/plugins/line-highlight/prism-line-highlight.js',
+        'media/js/syntax-prism.js'
+    ],
+    'search-suggestions': [
+        'media/js/search-suggestions.js'
+    ],
+    'wiki': [
+        'media/js/search-navigator.js',
+        'media/js/wiki.js'
+    ],
+    'wiki-edit': [
+        'media/js/wiki-edit.js',
+        'media/js/libs/tag-it.js',
+        'media/js/wiki-tags-edit.js'
+    ],
+    'wiki-move': [
+        'media/js/wiki-move.js'
+    ],
+    'newsletter': [
+        'media/js/newsletter.js'
+    ],
+    'html5shiv': [
+        'lib/js/html5shiv/dist/html5shiv.js'
+    ],
+    'jquery.hoverIntent': [
+        'lib/js/jquery.hoverIntent-1.5.0/index.js'
+    ]
+};
+
+gulp.task('default', ['build-javascript']);
+
+gulp.task('build-javascript', ['install-javascript-dependencies'], function() {
+    // Delete the old builds and build some new ones
+    del(jsBuildDirectory + '*-min-*.js', function() {
+        buildBundles(jsBundles, jsBuildDirectory, '.js');
+    });
+});
+
+gulp.task('install-javascript-dependencies', function() {
+    return gulp.src(jsDependencies)
+               .pipe(install());
+});
+
+/**
+ * Install any new requirements that were added to package.json and remove any
+ * requirements that are no longer mentioned, updating npm-shrinkwrap.json in
+ * the process.
+ *
+ * In other words, update node_modules and npm-shrinkwrap.json to reflect the
+ * current state of package.json.
+ */
+gulp.task('install-and-shrinkwrap-npm-dependencies', function() {
+    // There are Gulp plugins for some of these steps, but they don't work well
+    // together, so shell commands are used instead.
+    del('npm-shrinkwrap.json', function() {
+        gulp.src(npmDependencies)
+            .pipe(shell('npm prune'))
+            .pipe(shell('npm install'))
+            .pipe(shell('npm shrinkwrap --dev'));
+    });
+});
+
+gulp.task('watch', function() {
+    // NPM
+    gulp.watch(npmDependencies, ['install-and-shrinkwrap-npm-dependencies']);
+
+    // Bower
+    gulp.watch(jsDependencies, ['install-javascript-dependencies']);
+
+    // JavaScript
+    for(var bundleName in jsBundles) {
+        if(jsBundles.hasOwnProperty(bundleName)) {
+            var bundle = jsBundles[bundleName];
+            watchJSBundle(bundleName, bundle);
```

Comment:
Thanks for explaining that. I can't seem to reproduce locally. How long was the wait between steps 2 and 3? This sounds like [bug 1054257](https://bugzilla.mozilla.org/show_bug.cgi?id=1054257).

There are a few ways around that: we could fix [bug 1054257](https://bugzilla.mozilla.org/show_bug.cgi?id=1054257), tell developers to run Gulp on the host machine, or tell developers to wait for a few moments after making file changes. The first option could become a rabbit hole, but one of the others might work in the short term. On the plus side, the delay is no worse than the delay in compiling stylesheets, which we've become used to.